File: java/BillsApi/src/main/java/com/apilangarena/javabillsapi/infrastructure/persistence/repository/BillSummaryProjection.java
package com.apilangarena.javabillsapi.infrastructure.persistence.repository;

import java.math.BigDecimal;
import java.time.LocalDate;

public interface BillSummaryProjection {
    Long getId();

    String getBillNumber();

    LocalDate getIssuedAt();

    BigDecimal getTotal();

    String getCurrency();
}

File: kotlin/BillsApi/src/main/kotlin/com/apilangarena/billsapi/infrastructure/persistence/repository/BillSummaryProjection.kt
package com.apilangarena.billsapi.infrastructure.persistence.repository

import java.math.BigDecimal
import java.time.LocalDate

interface BillSummaryProjection {
    val id: Long
    val billNumber: String
    val issuedAt: LocalDate
    val total: BigDecimal
    val currency: String
}
